mwm-3-kotlin-1/presentation/PITCHME.md

```diff
@@ -1031,6 +1031,61 @@ public final class DateUtils {
 
 ---
 
+### <span style="color: #00B8D4; text-transform: none; font-size:0.8em;">Companion object</span> <span style="text-transform: none; font-size:0.8em;">of class</span>
+
+```kotlin
+class OtherActivity : Activity() {
+    
+    companion object {
+        private const val TAG = "OtherActivity"
+        fun startActivity(context: Context){
+            val intent = Intent(context, OtherActivity::class.java)
+            if(context !is Activity){
+                intent.flags = FLAG_ACTIVITY_NEW_TASK and FLAG_ACTIVITY_CLEAR_TASK
+            }
+            context.startActivity(intent)
+        }
+    }
+}
+```
+
+Note
+
+- In Kotlin, we don't have to write the companion object name.
+- `startActivity` method is called by calling `OtherActivity.Companion.startActivity` from Java.
+- `Companion` is the default name in case companion object doesn't have one.
+- We can write `companion object Fabric` and call method with `OtherActivity.Fabric.startActivity` in Java.
+
+---
+
+### <span style="color: #00B8D4; text-transform: none; font-size:0.8em;">Companion object</span> <span style="text-transform: none; font-size:0.8em;">of class</span>
+
+```java
+public final class OtherActivity extends Activity {
+   private static final String TAG = "OtherActivity";
+   public static final OtherActivity.Companion Companion = new OtherActivity.Companion((DefaultConstructorMarker)null);
+   public static final class Companion {
+      public final void startActivity(@NotNull Context context) {
+         Intrinsics.checkParameterIsNotNull(context, "context");
+         Intent intent = new Intent(context, OtherActivity.Companion.getClass());
+         if (!(context instanceof Activity)) {
+            intent.setFlags(0);
+         }
+         context.startActivity(intent);
+      }
+
+      private Companion() {
+      }
```

Comment:
You can maybe write that in the same line `private Companion() {}` (presentation)